build.sbt

```diff
@@ -691,6 +691,27 @@ lazy val benchmarks = project.module
     assembly / mainClass     := Some("org.openjdk.jmh.Main")
   )
 
+lazy val memoryLeakTests = project.module
+  .in(file("memory-leak-tests"))
+  .dependsOn(core.jvm, streams.jvm)
+  .settings(
+    crossScalaVersions --= List(Scala212, Scala3),
+    publish / skip           := true,
+    Test / parallelExecution := false,
+    Test / fork              := true,
+    Test / javaOptions       := List("-XX:+ExitOnOutOfMemoryError", "-Xmx1G", "-Xms200M"),
+    libraryDependencies ++= Seq(
+      "org.scalameta" %% "munit" % "1.0.0" % Test
```

Comment:
What's the reason for using `munit` over `zio-test`?

Reply:
For clean measurements, since memory leak test uses global heap usage for metrics. zio-test uses many ZIO components, including the operators we are testing. Practical measurements confirm this, with munit the scatter of values between measurements is much smaller.

For example, on my macbook, with munit:
```
Heap:    4_879_112 total
Heap:    4_884_800 total
Heap:    4_885_072 total
Heap:    4_884_608 total diff percent = 0.0
Heap:    4_901_088 total diff percent = 0.0
Heap:    4_902_728 total diff percent = 0.0
Heap:    4_902_704 total diff percent = 0.0
Heap:    4_902_544 total diff percent = 0.0
Heap:    4_902_128 total diff percent = 0.0
```

with zio-test (before last commit):
```
Heap:      7410728 total
Heap:      7517664 total
Heap:      7717104 total
Heap:      7870248 total diff percent = 6.0
Heap:      8075920 total diff percent = 8.0
Heap:      8275296 total diff percent = 11.0
Heap:      8427224 total diff percent = 13.0
Heap:      8636880 total diff percent = 16.0
Heap:      8832752 total diff percent = 19.0
```